### PR TITLE
storage: Show job progress in format dialogs and allow canceling

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Review by KKoukiou — COMMENTED** [review]
Cool feature. Do you think we can add a test for it? I tried it in the test vm, the cancel button gets enabled after some seconds of formating the storage, I click it, but the storage has been formated anyways. I guess the timing can be a bit tight. Let me know if it's possible to test this without relying too much on tight times. Otherwise I will approve.

**Comment by mvollmer** [comment]
> Do you think we can add a test for it?

It's difficult because this is inherently racy.  The Cancel button gets enabled and then after a while it gets disabled again (or the dialog closes), and the tests might just miss that window.

I'll try to reliably stop the erasing process somehow... maybe we can replace /bin/wipefs with something that blocks.

**Comment by mvollmer** [comment]
>  tried it in the test vm, the cancel button gets enabled after some seconds of formating the storage, I click it, but the storage has been formated anyways.

For manual testing, make sure to use some slow block device, such as a logical volume in a thin pool, and select "Overwrite all data with zeroes".  Then formatting should take some time and cancelling should leave "Unrecognized data" behind.

**Comment by mvollmer** [comment]
>  Do you think we can add a test for it?

Done.  Erasing doesn't block completely, but it is very very slow now.

File: test/verify/check-storage-format [hunk 1]
```diff
@@ -92,6 +92,39 @@ class TestStorageFormat(StorageCase):
         else:
             check_type("ntfs", 128)
 
+    def testFormatCandel(self):
```
Comment (KKoukiou) [inline test/verify/check-storage-format:95]: typo - s/candel/cancel

**Review by KKoukiou — CHANGES_REQUESTED** [review]
Sorry - one tiny typo and otherwise it's good to go.

oh - but it failed here https://logs.cockpit-project.org/logs/pull-14835-20201111-125625-ef162a97-ubuntu-2004/log.html#279

File: test/verify/check-storage-format (hunk 1 above)
Comment (mvollmer) [reply test/verify/check-storage-format:95]: You found it! :-) This typo was just too cute and seasonal to fix immediately, imo... :-)

Comment (mvollmer) [reply test/verify/check-storage-format:95]: Fixed.

**Comment by mvollmer** [comment]
> oh - but it failed here https://logs.cockpit-project.org/logs/pull-14835-20201111-125625-ef162a97-ubuntu-2004/log.html#279

Probably a flake.

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**